src/OVHSwiftAdapter.php

```diff
@@ -127,6 +127,48 @@ public function getTemporaryUrl(string $path, \DateTimeInterface $expiresAt, arr
         );
     }
 
+    public function getFormPostSignature(string $path, array $options = []): string
+    {
+        // Ensure Temp URL Key is provided for the Disk
+        if (empty($this->config->getTempUrlKey())) {
+            throw new \InvalidArgumentException("No Temp URL Key provided for container '".$this->container->name."'");
+        }
+
+        // expires is required otherwise a valid request will not be possible to be formed
+        if (!isset($options['expires'])) {
+            throw new \InvalidArgumentException("Missing 'expires' key in options");
+        }
+
+        // check if the 'expires' values is in the past
+        if (($options['expires'] ?? 0) < time()) {
+            throw new \InvalidArgumentException("Value of 'expires' cannot be in the past");
+        }
+
+        // Ensure $path doesn't begin with a slash
+        $path = ltrim($path, '/');
+
+        // Get the method
+        $method = 'POST';
```

Comment:
The method used for FromPost should be specified directly in the HTML Form or by the JS Ajax call type, there's no need to have it while generating a signature.

Reply:
As mentioned in the openstack [documentation](https://docs.openstack.org/swift/latest/api/form_post_middleware.html), it seems the method *must* be POST. We will have to update our documentation to reflect this.